addons/mail/controllers/bus.py

```diff
@@ -70,3 +70,12 @@ def mail_chat_history(self, uuid, last_id=False, limit=20):
             return []
         else:
             return channel._channel_fetch_message(last_id, limit)
+
+    @route('/longpolling/im_status', type="json", auth="user")
+    def im_status(self, partner_ids, guest_ids):
+        partners_im_status = super().im_status(partner_ids)
+        guest_im_status = request.env['mail.guest'].sudo().with_context(active_test=False).search([('id', 'in', guest_ids)]).read(['im_status'])
+        return {
+            'partners': partners_im_status,
+            'guests': guest_im_status,
+        }
```

Comment:
override should not change result format, the original needs to be changed to return a dict with partners as key then